lib/frameworks/README.md

```diff
@@ -17,33 +17,74 @@ exports.run = function(runner, specs)
-   jasmine/mocha, the `Scenario` in cucumber, or the method name in something
-   like jUnit.
-
- - `runner.runTestPreparer` must be called before any tests are run.
-
- - `runner.getConfig().onComplete` must be called when tests are finished.
-   It might return a promise, in which case `exports.run`'s promise should not
-   resolve until after `onComplete`'s promise resolves.
-
- - The returned promise must be resolved when tests are finished and it should return a results object. This object must have a `failedCount` property and optionally a `specResults`
- object of the following structure:
- ```
-   specResults = [{
-     description: string,
-     assertions: [{
-       passed: boolean,
-       errorMsg: string,
-       stackTrace: string
-     }],
-     duration: integer
-   }]
- ```
+- `runner.emit` must be called with `testPass` and `testFail` messages. These
+  messages must be passed a `testInfo` object, with a `name` and `category`
+  property. The `category` property could be the name of the `describe` block
+  in jasmine/mocha, the `Feature` in cucumber, or the class name in something
+  like jUnit. The `name` property could be the name of an `it` block in
+  jasmine/mocha, the `Scenario` in cucumber, or the method name in something
+  like jUnit.
+
+- `runner.runTestPreparer` must be called after the framework has been
+  initialized but before any spec files are run.  This function returns a
+  promise which should be waited on before executing tests.
+
+- `runner.getConfig().onComplete` must be called when tests are finished.
+  It might return a promise, in which case `exports.run`'s promise should not
+  resolve until after `onComplete`'s promise resolves.
+
+- The returned promise must be resolved when tests are finished and it should
+  return a results object. This object must have a `failedCount` property and
+  optionally a `specResults` object of the following structure:
+
+  ```ts
+  specResults: Array<{
+    description: string,
+    assertions: Array<{
+      passed: boolean,
+      errorMsg: string,
+      stackTrace: string
+    }>,
+    duration: integer
+  }>
+  ```
+
+
+### Future changes
+
+In Protractor 6.0, the requirements will be changed to the following:
+
+- `runner.afterEach` will have to be called after each test finishes.  It must
+  be called with a `testInfo` object of the following structure:
+
+  ```ts
+  testInfo: {
+    passed: boolean,
+    category: string,
+    name: string
+  }
+  ```
+
+  The `category` and `name` properties as the same as the ones that were in the
+  `testInfo` object from the old requirements.
+
+- `runner.beforeAll` must be called after the framework has been initialized but
+  before any spec files are run.  This function returns a promise which should
+  be waited on before executing tests.
+
+- The returned promise must follow the same rules as before.
+
+Note that `testPass` and `testFail` events will be replaced by
```

Comment:
We would still have the runner emit `testPass` and `testFail`, we would just do that in `runner.afterEach`.  So instead of having the framework handle event emission we'd do it ourselves.  And I think it's pretty redundant to have the frameworks do both.